### DISCUSSION TITLE
Reporting as files is poor design

### DISCUSSION THREAD

**Discussion by cigzigwon** [CMSgov/price-transparency-guide#679] [Q&A]
We are consuming large files but the influx of them with their flaws just sucks away time and resources.

Since this data is being sourced from databases it doesn't make sense to publish these files because nobody has technology to consume them nor audit them.

This would be much simpler for all parties if the data was provisioned as "Hypermedia based JSON". Easier to find and read data in this fashion and can still be provisioned statically on the web.

If I'm a consumer of a machine readable file and don't have the technology to consume it and once I get it find out that the data is un-readable... Who pays for that?

**Comment by cigzigwon** [comment]
eg. `https://example.com/cms_medicare/in-network-rates/provider_references/1`

Much simpler for me a consumer to leverage "hypermedia" w/linking to consume these references or any other context. It's much easier to build an API against a database then try to constantly get some performance gain out of streaming data.

And smaller businesses are never going to be able to consume these in a timely fashion. By time they get the data it will be outdated. This is not helping our economy as a whole.

**Comment by cigzigwon** [comment]
I'm calling my Senator.

**Comment by shaselton-usds** [comment]
These are raw data files in which APIs can be built on top of them in any fashion that makes the most sense to those that wish to provide an API, including hypermedia approaches. The raw data is a base starting point across industry in which anyone that is comfortable with large data sets can choose to manipulate, build, or search and build further services (again, such as APIs) to provide smaller subsets of the data for those further downstream to consume, which is sounds like where you might be at the moment.

**Reply by cigzigwon** [reply]
Since I'm creating MRFs to consume and needing to ingest them. I have insight to the actual effort required to produce and provide these datasets. It doesn't make much sense to need to try an address sharing large datasets across the web. It doesn't seem fair to inhibit and balloon costs of first that "is NOT comfortable with large data sets". They might be ill equipped. The problem you have here is needing to provide transparency only to turn around and charge customers more because of the large files and then compounded by nobody wanting to follow the rules like BCBS.  I'm .calling my Senator to report that all this federal regulation does is further centralize the market into the hands of power players and that it is only going to increase healthcare costs overall.

Since you aren't in the trenches witnessing these costs you are actually incorrect. The data becomes much easier to consume for all players via APIs. That is the whole point. It's also economics 101. All the providers need to do is copy out their data to a public realm with an API. It's much much easier to do then wield tools that don't even work with files over 100GB. All this does is duplicate data and I would argue that it's much much cheaper to cut this step because nobody can read these files into a system without hiring a data scientist.

I'm going to drive my point home with a Senator here in Michigan and after I've presented some examples I would like to hear any counterargument. Your recommendations so far have been a recipe for disaster. I care about my economy and you DO NOT.

**Reply by cigzigwon** [reply]
You get transparency at a higher health cost since we have to pay extra $$ to surface our duplicated data. We should be able to choose whether or not we want to duplicate our data in the first place. If we have 8M rate rows we have to copy them out to files instead of simply serving the data. This is nonsense.

**Reply by cigzigwon** [reply]
Looks like the criminals here have connections to BCBS/CIGNA. It's not the design to enable our economy. It's a design to bolster an oligarchy. Not on my watch!

**Reply by cigzigwon** [reply]
"Let's copy our data and spend thousands of dollars on doing so." Was that the overall goal here?? Seems like nobody thought about the actual problem here. It's a good design in theory but now a giant sink. I would hope to hear back from someone that is realistically having success with this model besides "too big to fail" and willing to burn money up.
